index.bs

```diff
@@ -137,55 +253,53 @@ available [=command|commands=] by module and name.
 end=] to the [=local end=], signaling that something of interest has
 occurred on the [=remote end=].
 
-Each concrete [=event=] type is defined by:
- - A string <dfn export for=event>name</dfn>, implicit in its
-   definition.
- - A <dfn export for=event>parameter type</dfn>, which is a |map| whose
-   entries are the details of the event that has occurred.
+ - A <dfn export for=event>event type</dfn> which is defined by a [=local
+   end schema=] fragment containing a group. Each such group has two fields:
+    - <code>method</code> which is a string literal of the form <code>[module
+      name].[event name]</code>. This is the <dfn export for=event>event
+      name</dfn>.
 
-Event [=event/parameter type|parameter types=] are defined in this
-specification using [[!RFC8610|CDDL]].
+    - <code>params</code> which defines a mapping containing event data. The
+      populated value of this map is the <dfn export for=command>event
+      parameters</code>.
 
 ## Processing Model ## {#processing-model}
 
-<div algorithm>
-To <dfn>process a command</dfn> given a qualified command name |qualified command name|,
-and parameters |parameters|:
-
-   1. If |qualified command name| does not contain exactly one U+002E
-      FULL STOP character (.), return an [=Error=] with [=error code=]
-      [=unknown command=].
 
-   1. Let |module name| be the bytes in |qualified command name|
-      preceeding the U+002E FULL STOP character.
+<div algorithm>
+To <dfn>process a command</dfn> given |data|:
 
-   1. Let |command name| be the bytes in |qualified command name|
-      following the U+002E FULL STOP character.
+   1. Match |data| against the [=remote end schema=]. If this results in a
+      match, run the following inner steps:
 
-   1. Let |command| be the command listed in the [=table of commands=]
-      whose module is |module name| and whose name is |command name|.
+      1. Let |result| be the result of running the [=remote end steps=]
+         for the command with the [=command name=] equal to the
+         <code>method</code> property of |data| and with the [=command
+         parameters=] from the matched data.
 
-   1. If there is no such |command|, return an [=Error=] with [=error
-      code=] [=unknown command=].
+      1. Assert: if |result| is a [=success=] its data matches the
+         schema for the [=result type=] corresponding to the command with name
+         [=command name=].
 
-   1. If |parameters| does not [=match a CDDL specification|match the
-      CDDL specification=] given by |command|'s [=command/parameter
-      type=], then return an [=Error=] with [=error code=] [=invalid
-      argument=].
+      1. Return <var>result</var>.
 
-   1. Let |result| be the result of [=trying=] to run the [=remote end
-      steps=] for |command| with |parameters|.
+   1. Otherwise there is no match. If |data| isn't a map or is a map without a
+      property named </code>method</code> return an [=Error=] with error code
```

Comment:
The wrong Error (https://w3c.github.io/webdriver/#dfn-errors) ends up linked here. Did you mean to refer to `Error` in the CDDL schema?

Reply:
No, this is the Error specification type defined in WebDriver rather than the `Error` CDDL definition.

Reply:
Aha. In that case, shouldn't it be lowercased?